### ISSUE TITLE
Modify coded tests to create resources in parallel

### ISSUE THREAD

**Issue by theunrepentantgeek** [Azure/azure-service-operator#3090]
**Describe the current behavior**

We've identified that some of our coded tests (in the `internal/controllers` package) create all the resources one at a time and therefore don't reveal whether there are issues with our sequencing/ordering of resource creation.

**Describe the improvement**

Instead of calling `CreateResourceAndWait()` one at a time, we should modify the tests to call `CreateResourcesAndWait()` (the plural form) to create everything at once. 

This may require rerecording the tests against live Azure, and may also flush out some edge case bugs we need to address.

**Copilot Instructions**

Scan tests in the `controllers` package to find cases where a single test makes multiple calls to `CreateResourceAndWait()` and there's a straightforward reordering of the code to group all the resource creations together into a single call. 

* If you spot any cases where the ARM ID of a resource is read from a newly created resource and used to construct a resource reference, modify that to use a within-cluster resource reference instead.
* No changes outside the `controllers` package are required.

**Comment by matthchr** [comment]
Would be good to do but low priority.

**Comment by theunrepentantgeek** [comment]
Still interested.

**Comment by matthchr** [comment]
Still interested in this, though many of the new ones do this by default, the old ones don't do it as well.

**Comment by matthchr** [comment]
Still interested in doing this - will see if copilot can help us here.

**Comment by theunrepentantgeek** [comment]
Closing as the change was too big to merge. See #5105 for a step-by-step approach.
